src/import/importConfiguration.ts

```diff
@@ -634,9 +634,17 @@ function parsePages(
   if (yamlPages == null) {
     return;
   }
-  return Object.entries(yamlPages).map(([nameUrl, details]) => {
-    return parsePage(nameUrl, details, `pages[${nameUrl}]`, file);
-  });
+  try {
+    return Object.entries(yamlPages).map(([nameUrl, details]) => {
+      return parsePage(nameUrl, details, `pages[${nameUrl}]`, file);
+    });
+  } catch (e) {
+    logger.error(
+      `SUSHI encountered the following error when parsing pages in ${file}. No page configuration will be used. `,
+      e
+    );
+    return [];
+  }
```

Comment:
I agree. I think adding a try/catch to every single parse method is way too much mess without much value. Your second comment does get at the difference here though. Because this was a recursive function, and because of the way the recursion works, when the function encountered the unexpected property in this configuration item, it caused the recursion to never end.

However, I took another look through this file to be sure I wasn't missing something else that could loop forever, and I confirmed there is only one other recursive parsing function in this file, `parseMenu`. The way this function is written, I believe it will always end.

So, instead of using the try/catch, I decided to update `parsePage` to also better ensure it will stop running in the case we encounter unexpected configurations. To do this, I made it so that `parsePage` is only called for `value`s that are objects. I believe this gets at what types of cases we want to recurse on. Previously, `parsePage` was being called on string `value`s, but then `Object.entries` will just continuously return an array in the shape `['0', 'h']`, but then it will be called again on `h`, which will return the same shape. This will no longer happen with a check for a type of object, which will fail on a string `'h'`. I wanted to update how this works though, because we would likely encounter the same issue if someone uses another unexpected property on the `pages` property in sushi-config.yaml in the future.

With this change, the two recursive functions in this file look like they will eventually end, so I don't think any try/catch (at the top or around individual functions) is needed.

Reply:
These changes are made in [0d76b61](https://github.com/FHIR/sushi/pull/1102/commits/0d76b61b0669a47993ca53b0c052fead9826c3f0). Let me know what you think.

Reply:
Good approach!